ui/app/pages/swaps/swaps-gas-customization-modal/swaps-gas-customization-modal.container.js

```diff
@@ -0,0 +1,154 @@
+import { connect } from 'react-redux'
+import {
+  hideModal,
+  setSwapsTxGasParams,
+} from '../../../store/actions'
+import {
+  conversionRateSelector as getConversionRate,
+  getCurrentCurrency,
+  getCurrentEthBalance,
+  getDefaultActiveButtonIndex,
+  getRenderableGasButtonData,
+} from '../../../selectors'
+
+import {
+  getSwapsCustomizationModalPrice,
+  getSwapsCustomizationModalLimit,
+  swapGasEstimateLoadingHasFailed,
+  swapGasPriceEstimateIsLoading,
+  getSwapGasPriceEstimateData,
+  swapCustomGasModalPriceEdited,
+  swapCustomGasModalLimitEdited,
+  isCustomSwapsGasPriceUnSafe,
+  getSwapsFallbackGasPrice,
+} from '../../../ducks/swaps/swaps'
+
+import {
+  addHexes,
+  getValueFromWeiHex,
+  sumHexWEIsToRenderableFiat,
+} from '../../../helpers/utils/conversions.util'
+import {
+  formatETHFee,
+} from '../../../helpers/utils/formatters'
+import {
+  calcGasTotal,
+  isBalanceSufficient,
+} from '../../send/send.utils'
+import SwapsGasCustomizationModalComponent from './swaps-gas-customization-modal.component'
+
+const mapStateToProps = (state) => {
+  const currentCurrency = getCurrentCurrency(state)
+  const conversionRate = getConversionRate(state)
+
+  const { modalState: { props: modalProps } = {} } = state.appState.modal || {}
+  const {
+    txParams = {},
+    customGasLimitMessage = '',
+    customTotalSupplement = '',
+    extraInfoRow = null,
+    initialGasPrice,
+    initialGasLimit,
+  } = modalProps || {}
+  const buttonDataLoading = swapGasPriceEstimateIsLoading(state)
+
+  const swapsCustomizationModalPrice = getSwapsCustomizationModalPrice(state)
+  const swapsCustomizationModalLimit = getSwapsCustomizationModalLimit(state)
+
+  const customGasPrice = swapsCustomizationModalPrice || initialGasPrice || getSwapsFallbackGasPrice(state)
```

Comment:
Is it possible for the `initialGasPrice` to be missing?

Reply:
I think if the fetch to the gas prices api failed it would be.

Reply:
It looks like `initialGasPrice` is set to `gasPrice` from `tradeTxParams`, which itself is set to `trade.gasPrice` if a custom gas price is not set. I'm not sure this has anything to do with the gas price API.

So this would be unset if `tradeTxParams` was unset, or if the trade returned from the API was missing a `gasPrice`.

I don't think it's possible to get here if `tradeTxParams` is unset (or shouldn't be at least! Things would be broken for other reasons in that case). Unfortunately I'm not sure whether `gasPrice` is guaranteed to be defined or not, since it's missing from the validator for the `trade` property of the quote object.

What I was going to say is that I suspect this fallback gas price is entirely unused, because the initial gas price is always set. But that depends on the API response.

Reply:
yeah, there is no `gasPrice` in the trade object of the api response

example api call https://metaswap-api.airswap-dev.codefi.network/trades?walletAddress=0xB0dA5965D43369968574D399dBe6374683773a65&slippage=3&sourceToken=0x6b175474e89094c44da98b954eedeac495271d0f&destinationToken=0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48&sourceAmount=1000000000000000000

Where is the best place to document that the trade object does not include `gasPrice`?

Reply:
Oh I see - it's set there after the fact in the swaps slice. I was confused because I saw `trade.gasPrice` being used later on.

Reply:
The correct places to document this would be a type - a JSDoc type annotation. In this case the object mutates though, which is unfortunate. It starts as a `trade` property of the quote, but then it starts getting used as `txParams` and additional properties are added to it.

Reply:
Maybe this is a task for another PR. Ideally I would like the type information to document where the property is or is not present, and that would require ensuring that the `quote.trade` object is distinct from `txParams` instead of being re-purposed as it is now.

Reply:
Okay, so it turns out you were right that the `getSwapsFallbackGasPrice` call is unnecessary in this file. Because it is already called in the selector that gets `tradeTxParams` in the swaps slice. I have removed this unnecessary call in f16e669ad

Reply:
I have created a ticket for further documentation on this: https://github.com/MetaMask/metamask-extension/issues/9712